### ISSUE TITLE
now dev seems to be "stuck" with a single configuration

### ISSUE THREAD

**Issue by brentkeller** [vercel/vercel#3184]
Windows 10
node v10.16.0
yarn v1.19.1
now v16.4.0

I seem to have gotten myself into a state where `now dev` will only successfully run for one specific configuration/set of dependencies. Once I add any new dependency using `yarn add` I get an error trying to execute `react-scripts start` when trying to run `now dev`, shown in the screenshot below. It doesn't seem to matter what the added dependency is. If I remove the dependency using `yarn remove` then `now dev` will work again. It seems like this one specific set of dependencies is all that will run.

![000024](https://user-images.githubusercontent.com/467397/67308803-c78c2d80-f4c8-11e9-8d96-8430d30080cf.png)

I decided to try running `now dev` against another project so I spun up a new create-react-app, added a dev script to run `react-scripts start` and found that `now dev` produces the same error.

I tried running a couple of the react sample projects from the `now-examples` repo and found that I get the same error with those as well.

I am able to successfully run `now dev` on all of these projects in an Ubuntu shell, which makes me think that there's something "stuck" about my `now` environment in Windows.

Things I've tried to get things working:
- Uninstalling and reinstalling different versions of `now` (using `npm i -g` with `now@latest`, `now@16.3.0`, `now@16.2.0`)
- Deleting my `node_modules` folder in my working directory
- Running `yarn cache clean`
- Deleting the folder `C:\Users\brent\AppData\Local\co.zeit.fun`
- Deleting the folder `C:\Users\brent\AppData\Local\co.zeit.now`
- Renaming/removing my `/api` folder to eliminate the node lambda builder
- Uninstall/reinstalling yarn
- Installed node.js 10.16.3 (using nvm), added yarn & now

Is there some other caching going on that I can try to clear or any other suggestions that I can try?

**Comment by leo** [comment]
Thank you for reporting this! I'll take a look. Reproducing it like this should work, right:

1. Running `npx create-react-app my-app`
2. Inside `package.json`, setting `dev` to `react-scripts start`
3. Running `now dev`
4. Installing a dependency using `yarn`
5. It breaks

Correct?

**Comment by brentkeller** [comment]
That's pretty much what I did. I created a couple of git branches along the way but I'm not sure what, if any, effect that had on things.

Let me know if there's any other info I can provide for you.

**Comment by rmariuzzo** [comment]
Something similar happens to me after running `now dev` for minutes. Any serverless function get stuck at:

> [debug] [2019-10-22T23:30:05.243Z] Searching files inside of /Users/a/b/c/

That search takes less than a second when it is not stuck. When stuck that search takes up to 2 mins...

**Comment by TooTallNate** [comment]
I can't seem to reproduce on a Windows box, following @leo's reproduction instructions. Is there any other details you can provide? In general the npm dependencies are not a responsibility of `now dev` and they are expected to be installed manually via `npm`/`yarn` before running `now dev`.

Can you let us know if `yarn run dev` gives the same "react-scripts is not recognized…" error message for you?

**Comment by brentkeller** [comment]
Calling `yarn start` and `yarn run dev` work as expected to launch the react server. Sorry for not including that in my original post.

`now dev` does work when I run it from my WSL Ubuntu shell so I was working from there last night and I managed to get the same "... not recognized..." error. I was able to solve it by fixing mismatched environment variable names in my `.env` and `now.json` files. I don't know if that's a clue or not because I'm not doing anything with env variables in the brand new CRA app or the now-examples apps.

Since I can't run any projects in Windows and it does work from the Ubuntu shell, I feel like that might rule out something in `node_modules` being the problem. This is part of what got me thinking that it might be something to do with a global cache or something in Windows/npm/yarn/now/something else.

Let me know if you have any suggestions or thoughts. I feel like the next step might be removing node.js and re-installing it.

**Comment by brentkeller** [comment]
I finally got enough down time to uninstall all versions of node then I installed 10.17.0, yarn and now. I'm able to run `now dev` with my project on different branches with different dependencies. I'm also having success with the `now-examples` create-react-app and another CRA test project. I guess something just got fouled up and uninstalling everything got things sorted. Sorry for the wild goose chase here.
